CardScan/Classes/CreditCardUtils.swift

```diff
@@ -40,68 +54,76 @@ public struct CreditCardUtils {
+        let cardNumberLength = cardNumber.count
         
-        return number.count == maxPanLength && (prefixesMastercard.contains(prefix2) || prefixesMastercard.contains(prefix3) || prefixesMastercard.contains(prefix4))
-    }
-    
-    public static func isVisa(number: String) -> Bool {
-        let prefix = String(number.prefix(1))
-        return number.count == maxPanLength && prefixesVisa.contains(prefix)
+        if cardNumber.isEmpty || network == CardNetwork.UNKNOWN {
+            return false
+        }
+        
+        switch network {
+        case CardNetwork.AMEX:
+            return cardNumberLength == maxPanLengthAmericanExpress
+        case CardNetwork.DINERSCLUB:
+            return cardNumberLength == maxPanLengthDinersClub
+        default:
+            return cardNumberLength == maxPanLength
+        }
     }
     
-    public static func isJcb(number: String) -> Bool {
-        let prefix = String(number.prefix(2))
-        return number.count == maxPanLength && prefixesJcb.contains(prefix)
+    public static func determineCardNetwork(cardNumber: String) -> CardNetwork {
+        let cardNumber = cardNumber.trimmingCharacters(in: .whitespaces)
+        
+        if cardNumber.isEmpty {
+            return CardNetwork.UNKNOWN
+        }
+        
+        switch true {
+        case hasAnyPrefix(cardNumber: cardNumber, prefixes: prefixesAmericanExpress):
+            return CardNetwork.AMEX
+        case hasAnyPrefix(cardNumber: cardNumber, prefixes: prefixesDiscover):
+            return CardNetwork.DISCOVER
+        case hasAnyPrefix(cardNumber: cardNumber, prefixes: prefixesJcb):
+            return CardNetwork.JCB
+        case hasAnyPrefix(cardNumber: cardNumber, prefixes: prefixesDinersClub):
+            return CardNetwork.DINERSCLUB
+        case hasAnyPrefix(cardNumber: cardNumber, prefixes: prefixesVisa):
+            return CardNetwork.VISA
+        case hasAnyPrefix(cardNumber: cardNumber, prefixes: prefixesMastercard):
+            return CardNetwork.MASTERCARD
+        case hasAnyPrefix(cardNumber: cardNumber, prefixes: prefixesUnionPay):
+            return CardNetwork.UNIONPAY
+        default:
+            return CardNetwork.UNKNOWN
+        }
     }
     
-    public static func isDinersClub(number: String) ->  Bool {
-        let prefix2 = String(number.prefix(2))
-        let prefix3 = String(number.prefix(3))
-        return number.count == maxPanLengthDinersClub && (prefixesDinersClub.contains(prefix2) || prefixesDinersClub.contains(prefix3))
+    static func hasAnyPrefix( cardNumber: String, prefixes: [String] ) -> Bool {
+        return prefixes.filter { cardNumber.hasPrefix($0) }.count > 0
     }
     
-    public static func format(number: String) -> String {
-        if number.count == 16 {
-            return format16(number: number)
-        } else if number.count == 15 {
-            return format15(number: number)
+    public static func format(cardNumber: String) -> String {
```

Comment:
if people are using this function they have to change their source code after this change, which we should avoid for changes like this